src/features/hover.zig

```diff
@@ -33,17 +34,27 @@ pub fn hoverSymbol(server: *Server, decl_handle: Analyser.DeclWithHandle, markup
             var buf: [1]Ast.Node.Index = undefined;
 
             if (tree.fullVarDecl(node)) |var_decl| {
+                if (var_decl.ast.type_node != 0)
+                    type_str = offsets.nodeToSlice(tree, var_decl.ast.type_node);
+
                 break :def Analyser.getVariableSignature(tree, var_decl);
             } else if (tree.fullFnProto(&buf, node)) |fn_proto| {
                 break :def Analyser.getFunctionSignature(tree, fn_proto);
             } else if (tree.fullContainerField(node)) |field| {
+                std.debug.assert(field.ast.type_expr != 0);
+                type_str = offsets.nodeToSlice(tree, field.ast.type_expr);
+
                 break :def Analyser.getContainerFieldSignature(tree, field);
             } else {
                 break :def Analyser.nodeToString(tree, node) orelse return null;
             }
         },
         .param_payload => |pay| def: {
             const param = pay.param;
+
+            std.debug.assert(param.type_expr != 0);
```

Comment:
Unfortunately I don't have a reproducer yet, but the following assertion failed while I was using your PR
```
thread 323332 panic: reached unreachable code
/opt/zig/0.11.0-dev.3886+0c1bfe271/files/lib/std/debug.zig:258:14: 0x32fb1c in assert (zls)
    if (!ok) unreachable; // assertion failure
             ^
/home/techatrix/zls/src/features/hover.zig:55:29: 0x48b547 in hoverSymbol (zls)
            std.debug.assert(param.type_expr != 0);
                            ^
/home/techatrix/zls/src/features/hover.zig:179:42: 0x3f29bd in hoverDefinitionGlobal (zls)
                .value = (try hoverSymbol(server, decl, markup_kind)) orelse return null,
                                         ^
/home/techatrix/zls/src/features/hover.zig:219:49: 0x3f1f40 in hover (zls)
        .var_access => try hoverDefinitionGlobal(server, source_index, handle),
                                                ^
/home/techatrix/zls/src/Server.zig:919:41: 0x39c3ea in hoverHandler (zls)
    const response = hover_handler.hover(server, source_index, handle);
                                        ^
/home/techatrix/zls/src/Server.zig:1506:35: 0x3600eb in processMessage (zls)
                break :blk handler(server, params) catch |err| {
                                  ^
/home/techatrix/zls/src/Server.zig:1362:26: 0x340d73 in processJsonRpc (zls)
    server.processMessage(message) catch |err| switch (message) {
                         ^
/home/techatrix/zls/src/main.zig:86:30: 0x34054a in loop (zls)
        server.processJsonRpc(json_message);
                             ^
/home/techatrix/zls/src/main.zig:428:13: 0x341828 in main (zls)
    try loop(server, record_file, replay_file);
            ^
/opt/zig/0.11.0-dev.3886+0c1bfe271/files/lib/std/start.zig:608:37: 0x32b0c4 in posixCallMainAndExit (zls)
            const result = root.main() catch |err| {
                                    ^
/opt/zig/0.11.0-dev.3886+0c1bfe271/files/lib/std/start.zig:367:5: 0x32abb1 in _start (zls)
    @call(.never_inline, posixCallMainAndExit, .{});
```